### ISSUE TITLE
route 53 changes detected even though there are no changes

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#361]
_This issue was originally opened by @tomstachura as hashicorp/terraform#10007. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

I submitted a similar problem report with the previous version and I think some of it was supposed to be solved, but I am still getting the following differences with the 7.7.10 version. 

The environment has just been created with apply and a plan has been run right after. It should be showing no changes, but it shows all the following:

```
~ module.tst1.aws_route53_record.r53_record
    alias.2190465367.evaluate_target_health: "true" => "false"
    alias.2190465367.name:                   "prod-web-1529429951.us-west-2.elb.amazonaws.com" => ""
    alias.2190465367.zone_id:                "Z1H1FL5HABSF5" => ""
    alias.2700809793.evaluate_target_health: "" => "true"
    alias.2700809793.name:                   "" => "PROD-WEB-1529429951.us-west-2.elb.amazonaws.com"
    alias.2700809793.zone_id:                "" => "Z1H1FL5HABSF5"

~ module.tst1.aws_security_group.admin
    ingress.#:                            "3" => "1"
    ingress.1938540428.cidr_blocks.#:     "1" => "0"
    ingress.1938540428.cidr_blocks.0:     "199.58.192.98/32" => ""
    ingress.1938540428.from_port:         "8091" => "0"
    ingress.1938540428.protocol:          "tcp" => ""
    ingress.1938540428.security_groups.#: "0" => "0"
    ingress.1938540428.self:              "false" => "false"
    ingress.1938540428.to_port:           "8091" => "0"
    ingress.3897505941.cidr_blocks.#:     "3" => "0"
    ingress.3897505941.cidr_blocks.0:     "199.58.192.98/32" => ""
    ingress.3897505941.cidr_blocks.1:     "35.160.166.68/32" => ""
    ingress.3897505941.cidr_blocks.2:     "35.161.227.236/32" => ""
    ingress.3897505941.from_port:         "22" => "0"
    ingress.3897505941.protocol:          "tcp" => ""
    ingress.3897505941.security_groups.#: "0" => "0"
    ingress.3897505941.self:              "false" => "false"
    ingress.3897505941.to_port:           "22" => "0"
    ingress.4282511762.cidr_blocks.#:     "1" => "1"
    ingress.4282511762.cidr_blocks.0:     "10.5.101.236/32" => "10.5.101.236/32"
    ingress.4282511762.from_port:         "22" => "22"
    ingress.4282511762.protocol:          "tcp" => "tcp"
    ingress.4282511762.security_groups.#: "0" => "0"
    ingress.4282511762.self:              "false" => "false"
    ingress.4282511762.to_port:           "22" => "22"

~ module.tst1.lb_id.aws_route53_record.r53_record
    alias.1412314257.evaluate_target_health: "true" => "false"
    alias.1412314257.name:                   "prod-id-1538439218.us-west-2.elb.amazonaws.com" => ""
    alias.1412314257.zone_id:                "Z1H1FL5HABSF5" => ""
    alias.1713183629.evaluate_target_health: "" => "true"
    alias.1713183629.name:                   "" => "PROD-ID-1538439218.us-west-2.elb.amazonaws.com"
    alias.1713183629.zone_id:                "" => "Z1H1FL5HABSF5"

~ module.tst1.lb_solr.aws_route53_record.r53_record
    alias.1351171064.evaluate_target_health: "" => "true"
    alias.1351171064.name:                   "" => "internal-PROD-SOLR-236462827.us-west-2.elb.amazonaws.com"
    alias.1351171064.zone_id:                "" => "Z1H1FL5HABSF5"
    alias.907506756.evaluate_target_health:  "true" => "false"
    alias.907506756.name:                    "internal-prod-solr-236462827.us-west-2.elb.amazonaws.com" => ""
    alias.907506756.zone_id:                 "Z1H1FL5HABSF5" => ""

~ module.tst1.lb_ui.aws_route53_record.r53_record
    alias.2037855967.evaluate_target_health: "" => "true"
    alias.2037855967.name:                   "" => "PROD-UI-85055216.us-west-2.elb.amazonaws.com"
    alias.2037855967.zone_id:                "" => "Z1H1FL5HABSF5"
    alias.884876570.evaluate_target_health:  "true" => "false"
    alias.884876570.name:                    "prod-ui-85055216.us-west-2.elb.amazonaws.com" => ""
    alias.884876570.zone_id:                 "Z1H1FL5HABSF5" => ""

~ module.tst1.lb_web.aws_route53_record.r53_record
    alias.2190465367.evaluate_target_health: "true" => "false"
    alias.2190465367.name:                   "prod-web-1529429951.us-west-2.elb.amazonaws.com" => ""
    alias.2190465367.zone_id:                "Z1H1FL5HABSF5" => ""
    alias.2700809793.evaluate_target_health: "" => "true"
    alias.2700809793.name:                   "" => "PROD-WEB-1529429951.us-west-2.elb.amazonaws.com"
    alias.2700809793.zone_id:                "" => "Z1H1FL5HABSF5"
```

**Comment by Jonnymcc** [comment]
Still happens in `Terraform v0.11.2 + provider.aws v1.7.1`

**Comment by bflad** [comment]
I have submitted a PR to hopefully address this for uppercase named ALB/ELB DNS alias: #3119

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released in v1.8.0 of the provider, expected tomorrow! 🎉

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
